src/maker/api.rs

```diff
@@ -510,9 +531,15 @@ pub fn check_for_broadcasted_contracts(maker: Arc<Maker>) -> Result<(), MakerErr
                             "[{}] Spawning recovery thread after seeing contracts in mempool",
                             maker.config.port
                         );
-                        let handle = std::thread::spawn(move || {
-                            recover_from_swap(maker_clone, outgoings, incomings).unwrap();
-                        });
+                        let handle = std::thread::Builder::new()
+                            .name("recovery: saw contracts in mempool".to_string())
+                            .spawn(move || {
+                                if let Err(e) =
+                                    recover_from_swap(maker_clone.clone(), outgoings, incomings)
```

Comment:
nit: 
``` suggestion
                                    recover_from_swap(maker_clone, outgoings, incomings)
```
No need to clone again when we already clone the `maker` to get `maker_clone` for recovery thread usage.